pkg/apis/core/types_shoot.go

```diff
@@ -600,6 +615,30 @@ type KubeletConfigEvictionSoftGracePeriod struct {
 	NodeFSInodesFree *metav1.Duration
 }
 
+// KubeletConfigKubeReserved contains the kube-reserved resources configuration
+type KubeletConfigKubeReserved struct {
+	// CPU is the kube-reserved cpu.
+	CPU *resource.Quantity
+	// Memory is the kube-reserved memory.
+	Memory *resource.Quantity
+	// EphemeralStorage is the kube-reserved ephemeral-storage.
+	EphemeralStorage *resource.Quantity
+	// PID is the kube-reserved process-ids.
+	PID *int64
+}
+
+// KubeletConfigSystemReserved contains the system-reserved resources configuration
+type KubeletConfigSystemReserved struct {
+	// CPU is the system-reserved cpu.
+	CPU *resource.Quantity
+	// Memory is the system-reserved memory.
+	Memory *resource.Quantity
+	// EphemeralStorage is the system-reserved ephemeral-storage.
+	EphemeralStorage *resource.Quantity
+	// PID is the system-reserved process-ids.
+	PID *int64
+}
```

Comment:
Can we make those things the same as the `kubelet`[config](https://github.com/kubernetes/kubernetes/blob/9016740a6ffe91bb29824f80c34087b993903bd6/pkg/kubelet/apis/kubeletconfig/v1beta1/types.go#L618-L645) (being `map[string]string`)?

Reply:
Well, this implementation is consistent with the existing gardener examples for [kubelet params](https://github.com/gardener/gardener/blob/master/pkg/apis/core/types_shoot.go#L562)  which are also modeled as `map[string]string` in [kubelet apis](https://github.com/kubernetes/kubernetes/blob/9016740a6ffe91bb29824f80c34087b993903bd6/pkg/kubelet/apis/kubeletconfig/v1beta1/types.go#L486). It also makes validation convenient.

I'm willing to make the change, but can you explain why you prefer this approach? Do you have future plans to embed the k8s kubeletconfig type in the worker spec? If so, do you think that validation is not required here, and it's fine to just let kubelet validate the config?

Reply:
Agree with @guydaichs , convenient validation is the reason why we are using concrete types. 
I would opt for keeping it the way it is. But lets wait for @mvladev response